jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/SslProperties.java

```diff
@@ -0,0 +1,349 @@
+    private static final String DEFAULT_TRUSTSTORE_TYPE = "JKS";
+
+    @Builder.Default
+    private final SslMode sslMode = SslMode.DEFAULT_TLS;
+
+    // For SslMode.ONE_SIDED_TLS and SslMode.MUTUAL_TLS (optional for both)
+    @Builder.Default
+    private final String truststorePathValue = null;
+
+    // For SslMode.ONE_SIDED_TLS and SslMode.MUTUAL_TLS (optional for both, only if truststore is used)
+    @Builder.Default
+    private final String truststorePasswordValue = null;
+
+    // For SslMode.ONE_SIDED_TLS and SslMode.MUTUAL_TLS (optional for both, only if truststore is used)
+    @Builder.Default
+    private final String truststoreTypeValue = DEFAULT_TRUSTSTORE_TYPE;
+
+    // For SslMode.MUTUAL_TLS (required for mutual TLS)
+    @Builder.Default
+    private final String clientCertPathValue = null;
+
+    // For SslMode.MUTUAL_TLS (required for mutual TLS)
+    @Builder.Default
+    private final String clientKeyPathValue = null;
+
+    // For SslMode.ONE_SIDED_TLS and SslMode.MUTUAL_TLS (optional for both, alternative to truststore)
+    @Builder.Default
+    private final String caCertPathValue = null;
+
+    /**
+     * Parses connection properties from a Properties object.
+     * Removes the interpreted properties from the Properties object.
+     *
+     * @param props The properties to parse
+     * @return A SslProperties instance
+     * @throws SQLException if parsing of property values fails
+     */
+    public static SslProperties ofDestructive(Properties props) throws SQLException {
+        SslPropertiesBuilder builder = SslProperties.builder();
+
+        // Extract ssl.disabled early (needed for validation and when ssl.disabled is passed explicitly as false)
+        boolean sslDisabled =
+                Boolean.parseBoolean(takeOptional(props, SSL_DISABLED).orElse("false"));
+
+        if (sslDisabled) {
+            // checks if any additional properties are present with sslDisabled, if yes, then it throws error
+            if (!Collections.disjoint(
+                    props.keySet(),
+                    Arrays.asList(
+                            SSL_CLIENT_CERT_PATH,
+                            SSL_CLIENT_KEY_PATH,
+                            SSL_CA_CERT_PATH,
+                            SSL_TRUSTSTORE_PATH,
+                            SSL_TRUSTSTORE_PASSWORD,
+                            SSL_TRUSTSTORE_TYPE))) {
+                throw new SQLException("Cannot specify ssl.disabled=true with other SSL properties. ", "HY000");
+            }
+            builder.sslMode(SslMode.DISABLED);
+        } else if (props.containsKey(SSL_CLIENT_CERT_PATH) && props.containsKey(SSL_CLIENT_KEY_PATH)) {
+            // if both client cert and client key are present, TLS mode should be mtls
+            if (props.containsKey(SSL_CA_CERT_PATH) && props.containsKey(SSL_TRUSTSTORE_PATH)) {
+                throw new SQLException("Cannot specify both ssl.ca.certPath and ssl.truststore.path. ", "HY000");
+            }
```

Comment:
logically, this check belongs further down towards "Either ca_cert or Trust store can be present,not both"